react-components/src/architecture/base/renderTarget/CommandsController.test.ts

```diff
@@ -0,0 +1,498 @@
+      controller.setActiveTool(activeTool);
+
+      for (const isLeftButton of [false, true]) {
+        const mock = vi.fn();
+        if (isLeftButton) {
+          activeTool.onLeftPointerDown = mock;
+        } else {
+          defaultTool.onRightPointerDown = mock;
+        }
+        await controller.onPointerDown(createMouseDownEvent(isLeftButton), isLeftButton);
+        expect(mock).toHaveBeenCalledTimes(1);
+      }
+    });
+
+    test('Should call correction function on the correct tool on mouse drag', async () => {
+      const defaultTool = createMockTool2();
+      const activeTool = createMockTool1();
+      controller.setDefaultTool(defaultTool);
+      controller.setActiveTool(activeTool);
+
+      for (const isLeftButton of [false, true]) {
+        const mock = vi.fn();
+        if (isLeftButton) {
+          activeTool.onLeftPointerDrag = mock;
+        } else {
+          defaultTool.onRightPointerDrag = mock;
+        }
+        await controller.onPointerDrag(createMouseMoveEvent(isLeftButton), isLeftButton);
+        expect(mock).toHaveBeenCalledTimes(1);
+      }
+    });
+
+    test('Should call correction function on the correct tool on mouse up', async () => {
+      const defaultTool = createMockTool2();
+      const activeTool = createMockTool1();
+      controller.setDefaultTool(defaultTool);
+      controller.setActiveTool(activeTool);
+
+      for (const isLeftButton of [false, true]) {
+        const mock = vi.fn();
+        if (isLeftButton) {
+          activeTool.onLeftPointerUp = mock;
+        } else {
+          defaultTool.onRightPointerUp = mock;
+        }
+        await controller.onPointerUp(createMouseUpEvent(isLeftButton), isLeftButton);
+        expect(mock).toHaveBeenCalledTimes(1);
+      }
+    });
+  });
+
+  function createMockTool1(): MockTool1 {
+    const tool = new MockTool1();
+    tool.attach(renderTarget);
+    return tool;
+  }
+  function createMockTool2(): MockTool2 {
+    const tool = new MockTool2();
+    tool.attach(renderTarget);
+    return tool;
+  }
+});
+
+class MockTool1 extends BaseTool {
+  // Mock any methods or properties as needed
+}
+class MockTool2 extends BaseTool {}
+
+class MockCommand extends BaseCommand {
+  protected override get shortCutKey(): string | undefined {
+    return 'A';
+  }
+
+  protected override get shortCutKeyOnAlt(): boolean {
+    return true;
+  }
+
+  protected override get shortCutKeyOnCtrl(): boolean {
+    return true;
+  }
+
+  protected override get shortCutKeyOnShift(): boolean {
+    return true;
+  }
+}
+
+export class UpdateTester {
+  public CalledTimes = 0;
+
+  // Set isCalled to true if the change is detected
+  public constructor(command: BaseCommand, change?: symbol) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    function listener(_command: BaseCommand, inputChange?: symbol): void {
+      if (change === inputChange) {
+        self.CalledTimes++;
+      }
```

Comment:
I think you would get rid of this alias if you declare `listener` as `const listener = (...): void => { ... }`, if you still want to keep this class